src/ButtonGroup.js

```diff
@@ -0,0 +1,45 @@
+import styled from 'styled-components'
+import Box from './Box'
+
+const ButtonGroup = styled(Box)`
+  vertical-align: middle;
+
+  & > * {
+    position: relative;
+    border-right-width: 0;
+    border-radius: 0;
+
+    :first-child {
+      border-top-left-radius: 3px;
+      border-bottom-left-radius: 3px;
+    }
+
+    :last-child {
+      border-right-width: 1px;
+      border-top-right-radius: 3px;
+      border-bottom-right-radius: 3px;
+    }
+
+    :focus,
+    :active,
+    :hover {
+      border-right-width: 1px;
+      + * {
+        border-left-width: 0;
+      }
+    }
+
+    :focus,
+    :active {
+      z-index: 1;
+    }
+  }
+`
+ButtonGroup.defaultProps = {
+  display: 'inline-block'
```

Comment:
If you set `display: inline-block` in the styled components definition, does the system prop not work?

Reply:
Yeah, it overrides it because the styled-system prop gets added first